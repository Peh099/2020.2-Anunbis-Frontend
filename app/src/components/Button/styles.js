import styled from 'styled-components';

export const Btn = styled.button`
    background: ${props => props.backColor ? props.backColor : ""};
    background-image: url(${props => props.backImage ? props.backImage : ""});
    border: 1px solid rgba(255, 245, 157, 0.6);
    box-sizing: border-box;
    border-radius: ${props => props.radius ? props.radius : "10px"};
    padding: ${props => props.padding ? props.padding : '15px 10px'};
    margin-inline: 15px;
    
    &:focus{
        outline-width: 0;
    }
    
    &:hover{
        cursor: ${props => props.cursor ? props.cursor : 'pointer'};
    }
`