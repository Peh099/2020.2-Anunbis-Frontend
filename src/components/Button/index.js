import React from "react";
import styled from 'styled-components';

const Btn = styled.button`
    background: ${props => props.backColor ? props.backColor : ""};
    border: 1px solid rgba(255, 245, 157, 0.6);
    box-sizing: border-box;
    border-radius: ${props => props.radius ? props.radius : "10px"};
    padding: ${props => props.padding ? props.padding : '15px 10px'};
    margin-inline: 15px;
    box-shadow: 2px 2px grey;
    
    &:focus{
        outline-width: 0;
    }
    
    &:hover{
        cursor: ${props => props.cursor ? props.cursor : 'pointer'};
    }
`  

export default function Button(props) {
    return (
            <Btn type={props.type} onClick={props.onClick} backColor={props.backColor} padding={props.padding} radius={props.radius} className={props.className} cursor={props.cursor}>
                {props.text}
            </Btn>
    );
}