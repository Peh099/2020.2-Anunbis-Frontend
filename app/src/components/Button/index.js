import React from 'react';

import { Btn } from './styles';

export default function Button(props) {
  return (
    <Btn {...props}>
      {props.text}
    </Btn>
  );
}
