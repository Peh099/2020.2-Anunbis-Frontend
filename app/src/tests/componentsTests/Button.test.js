import renderer from 'react-test-renderer';
import Button from '../../components/Button';

describe('Snapshot Button component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
