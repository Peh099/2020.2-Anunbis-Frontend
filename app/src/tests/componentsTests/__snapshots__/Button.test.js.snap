// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Button component matches the snapshot 1`] = `
<button
  className="sc-bdnxRM bSShnC"
/>
`;
